express/scripts/preference-store.js

```diff
@@ -23,26 +23,72 @@
  * /express/blocks/feature-grid-desktop/feature-grid-desktop.js#L147
  */
 
+/** pre-defining all the accepted eventNames */
+export const preferenceNames = {
+  reduceMotion: {
+    name: 'reduceMotion',
+    mediaQueryString: '(prefers-reduced-motion: reduce)',
+  },
+};
+
 class PreferenceStore {
   constructor() {
     this.stores = {}; // { [name]: { value, subscribers: [HTMLElement] } }
   }
 
+  updateStore(name) {
+    const value = localStorage.getItem(name) === 'on';
+
+    if (!this.stores[name]) this.stores[name] = { subscribers: [] };
+    const store = this.stores[name];
+
+    store.value = value;
+    store.subscribers.forEach((sub) => {
+      sub.dispatchEvent(new CustomEvent(name, { detail: store }));
+    });
+    return store;
+  }
+
   /**
    * Sample usage:
    * import preferenceStore, { eventNames } from '../../scripts/preference-store.js';
-   * preferenceStore.set(eventNames.reduceMotion, sessionStorage.getItem('reduceMotion'));
+   * preferenceStore.init('reduceMotion');
    */
-  set(name, value) {
-    if (!this.stores[name]) this.stores[name] = { subscribers: [] };
-    const store = this.stores[name];
+  init(name) {
+    const mediaQuery = window.matchMedia(preferenceNames[name].mediaQueryString);
+    const syncBrowserSetting = () => {
+      if (mediaQuery === true || mediaQuery.matches === true) {
+        localStorage.setItem(name, 'on');
+      } else {
+        localStorage.setItem(name, 'off');
+      }
+    };
 
-    if (value === this.get(name)) return;
+    if (mediaQuery === true || mediaQuery.matches === true) {
+      syncBrowserSetting();
+    }
 
-    store.value = value;
-    store.subscribers.forEach((sub) => {
-      sub.dispatchEvent(new CustomEvent(name, { detail: value }));
+    mediaQuery.addEventListener('change', () => {
```

Comment:
being opinionated, I prefer to have things modifying the store live outside of the store, just so we keep a clear mental model that the store only stores things, and it's the owner/setter or customers who come here to get & set.